src/flags-fetch.ts

```diff
@@ -78,7 +79,12 @@ export async function fetchFlags(url: string, timeoutMs: number) {
     try {
       const controller = new AbortController();
       const id = setTimeout(() => controller.abort(), timeoutMs);
-      const res = await fetch(url, {
+
+      // add SDK version to the query params
+      const separator = url.includes("?") ? "&" : "?";
+      const newUrl = `${url}${separator}${SDK_VERSION_HEADER_NAME}=${SDK_VERSION}`;
+
+      const res = await fetch(newUrl, {
```

Comment:
String concatenation for urls always leads to poblems. Let's use the url object for safe manipulation instead.

```suggestion
      const urlObj = new URL(url);
      urlObj.set(SDK_VERSION_HEADER_NAME, SDK_VERSION);

      const res = await fetch(urlObj, {
```

Reply:
nice!